### ISSUE TITLE
Is Backdrop's required PHP version still correct?

### ISSUE THREAD

**Issue by krueschi** [backdrop/backdrop-issues#3020]
## Describe your issue or idea

Readme.md file states as minimal php version 5.3.2 as requirement. So is this version number wrong?

### Steps to reproduce (if reporting a bug)

- Copy backdrop 1.9.3 code to a webserver with php 5.3.29.
- Enter in your database credentials
- Installation process starts as normal and runs up to site information form. After hitting the save button the screen turns white.

### Actual behavior (if reporting a bug)

Trying to install backdrop cms with php 5.3.29 failed with a white screen.

### Expected behavior (if reporting a bug)

Finishing the installation should show the vanilla site.

### Relevant version/system information (if applicable)

After some time tinkering around I found that updating php to 7.1.7 does the job: installation process runs just fine and after finishing it displays the vanilla site as expected.

**Comment by opi** [comment]
I agree that we need to fix the README ; Also, Backdrop should not recommend to use an outdated PHP version. PHP 5.6 is the minimum at this time, see https://secure.php.net/supported-versions.php

**Comment by krueschi** [comment]
So should we then test which versions are valid down to 5.6 or will we recommend php 7.x anyway?

**Comment by olafgrabienski** [comment]
> Readme.md file states as minimal php version 5.3.2 as requirement. So is this version number wrong?

I don't think that the version number is wrong. I know of a Backdrop installation which runs fine on PHP 5.3.3, and as far as I know, the PHP requirements haven't changed recently. The failure of the installation process may have had other reasons, for instance some PHP settings in your 5.3.29 version.

@krueschi Are you up to do further research on the installation failure reason, or are you happy with your update to PHP 7.1.x?

**Comment by krueschi** [comment]
> Are you up to do further research on the installation failure reason, or are you happy with your update to PHP 7.1.x?

I'm fine with php 7.1.x but if there is no urge I'm willing to do some tests to evaluate if its php 5.3.29 fault, a php settings thing or because of a php version lower than 5.6 / 7.1 in general.

**Comment by olafgrabienski** [comment]
> Also, Backdrop should not recommend to use an outdated PHP version.

I agree that Backdrop shouldn't recommend outdated PHP versions. So, let's encourage the use of newer versions in README and on the [Requirements page](https://backdropcms.org/requirements), without stopping support for the outdated versions though.

**Comment by krueschi** [comment]
So what would be the best practice? Providing a patch / pull request for a new README.md after testing with
a) a minimal php statement and
b) an at-least-recommend php version?

**Comment by olafgrabienski** [comment]
Generally, a pull request would be good! I'm however not sure how we would stay up to date regarding the recommended php version.

Regarding the minimum version, without further indications, I wouldn't suspect that the requirements have changed. And tests may be difficult as they had to cover a lot of cases. In a recent comment of @jenlampton I've read for example that the minimum PHP version had been changed to PHP 5.3.2 (in comparison to Drupal 7 needing only 5.2.4) because of a Date module's requirement, see https://github.com/backdrop/backdrop-issues/issues/2967#issuecomment-365444377.

**Comment by jenlampton** [comment]
I still use backdrop regularly on php 5.3.29 - as it's part of my upgrade process from Drupal 6 -  Backdrop core does work on this PHP version. (But I have found that some Backdrop contrib modules do not.)

I think it would still be good to update the README to recommend that people use the latest supported version of PHP, regardless of Backdrop's minimum version, but I would not recommend bumping the minimum version up any further, simply because there are still people still upgrading from Drupal 6, and those servers have old PHP versions on them.

@krueschi it may be that the installation problems are experiencing are related to something other than PHP version. It would be good to figure out what it is and add a check for that into the installer so that installation will not begin until all issues are resolved.

**Comment by opi** [comment]
> I still use backdrop regularly on php 5.3.29 - as it's part of my upgrade process from Drupal 6 - Backdrop core does work on this PHP version.

Current requirement: **"PHP 5.3.2 or higher."**

Rewording proposal: **"PHP 5.3.2 or higher. We recommend you to use at least PHP 5.6 which is the latest supported version."**

On the b.org version, "supported version" could link to https://secure.php.net/supported-versions.php

I'm not a native english speaker, and it's just a proposal ;)

**Comment by jenlampton** [comment]
I think we should exclude `PHP 5.6` so that we don't need to update that number as PHP ages. How about: 
```
PHP 5.3.2 or higher. 
Even though Backdrop can run on older versions of PHP, we strongly recommend that you use a supported version of PHP:  https://secure.php.net/supported-versions.php.
```

**Comment by olafgrabienski** [comment]
> PHP 5.3.2 or higher.
> Even though Backdrop can run on older versions of PHP, we strongly recommend ...

Not sure if the "older versions" part could be mistaken as "even older versions than PHP 5.3.2".

Hopefully more clear:

> Even though Backdrop can run on older versions like PHP 5.3.2 or 5.4, we strongly recommend ...

**Comment by krueschi** [comment]
@krueschi it may be that the installation problems are experiencing are related to something other than PHP version. It would be good to figure out what it is and add a check for that into the installer so that installation will not begin until all issues are resolved.

As the problematic server installation was on a shared server of the hosting company I will try to duplicate its configuration locally and test before I will report here.

**Comment by laryn** [comment]
I wonder if talking about "supported versions" could confuse people, because we're not talking about versions that Backdrop supports, but versions that the PHP project supports. Throwing one more variation out there that might make that a little more clear without being too wordy:

> PHP 5.3.2 or higher (An [actively supported version of PHP](https://secure.php.net/supported-versions.php) is strongly recommended)

**Comment by herbdool** [comment]
I think the PR is clear since it's got a link to show which versions specifically are recommended.

**Comment by opi** [comment]
:+1: This is minimal effort, by just adding a link to PHP official documentation. Not sure that we have to bikeshed anymore here. I'd rather debate if *Backdrop should encourage the use of PHP5.6, for security reasons*, but in another issue.

**Comment by quicksketch** [comment]
Great, I've merged https://github.com/backdrop/backdrop/pull/2120 into 1.x and 1.10.x.

Another "just FYI" item here as to why we still support PHP 5.3: CentOS/RHEL 6 uses 5.3.2 by default, and theoretically is backporting security fixes into PHP until 2020. Although they are the last major distribution to support older versions of PHP in a supported OS version.

Though honestly I think @jenlampton's reason is better: it makes going from Drupal 6 to Backdrop a lot easier if you can run both sites on the same PHP version.
